e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/report-reconciliation/bank-utilisation-report-page.spec.js

```diff
@@ -0,0 +1,118 @@
+import {
+  CURRENCY,
+  FeeRecordEntityMockBuilder,
+  REPORT_NOT_RECEIVED,
+  RECONCILIATION_IN_PROGRESS,
+  UtilisationReportEntityMockBuilder,
+  toIsoMonthStamp,
+  getPreviousReportPeriodForBankScheduleByMonth,
+  FEE_RECORD_STATUS,
+  getFormattedReportPeriodWithLongMonth,
+} from '@ukef/dtfs2-common';
+import pages from '../../../pages';
+import USERS from '../../../../fixtures/users';
+import { NODE_TASKS } from '../../../../../../e2e-fixtures';
+
+const { utilisationReportPage } = pages;
+
+context('Bank utilisation report page', () => {
+  const SUBMISSION_MONTH = toIsoMonthStamp(new Date());
+  const BANK_ID = '961';
+  const REPORT_ID = 1;
+  const FEE_RECORD_ID_ONE = '11';
+  const FACILITY_ID_ONE = '11111111';
+  let bankName;
+  let reportPeriodString;
+
+  const aTfmFacilityFacilitySnapshot = () => ({
+    coverStartDate: new Date().getTime(),
+    coverEndDate: new Date().getTime(),
+    interestPercentage: 5,
+    dayCountBasis: 365,
+    coverPercentage: 80,
+    value: 100000,
+  });
+
+  const TFM_FACILITIES = [
+    {
+      facilitySnapshot: { ...aTfmFacilityFacilitySnapshot(), ukefFacilityId: FACILITY_ID_ONE },
+    },
+  ];
+
+  before(() => {
+    const visibleBanks = [];
+
+    cy.task(NODE_TASKS.DELETE_ALL_TFM_FACILITIES_FROM_DB);
+    cy.task(NODE_TASKS.INSERT_TFM_FACILITIES_INTO_DB, TFM_FACILITIES);
+
+    cy.task(NODE_TASKS.GET_ALL_BANKS).then((getAllBanksResult) => {
+      getAllBanksResult
+        .filter((bank) => bank.isVisibleInTfmUtilisationReports)
+        .forEach((visibleBank) => {
+          visibleBanks.push(visibleBank);
+        });
+    });
+
+    cy.task(NODE_TASKS.DELETE_ALL_FROM_SQL_DB);
+
+    cy.wrap(visibleBanks).each((bank) => {
+      const reportPeriod = getPreviousReportPeriodForBankScheduleByMonth(bank.utilisationReportPeriodSchedule, SUBMISSION_MONTH);
+
+      reportPeriodString = getFormattedReportPeriodWithLongMonth(reportPeriod);
```

Comment:
This is getting overridden for each bank, and so if the last bank was actually not the one we are navigating to and has a different reporting schedule e.g. quarterly not monthly, then this will fail. This can happen if someone adds a new bank for example to mock data loader.

For this test to be reliable to changes to mock-data-loader we should only set this inside the if block starting on line 63.

Reply:
changed